packages/aws-cdk/test/api/hotswap/state-machine-hotswap-deployments.test.ts

```diff
@@ -0,0 +1,478 @@
+  // GIVEN
+  setup.setTemplate({
+    Resources: {
+      Machine: {
+        Type: 'AWS::StepFunctions::StateMachine',
+        Properties: {
+          DefinitionString: {
+            'Fn::Join': [
+              '',
+              [
+                '{ "Prop" : "old-value" }',
+              ],
+            ],
+          },
+        },
+      },
+    },
+  });
+  const cdkStackArtifact = setup.cdkStackArtifactOf({
+    template: {
+      Resources: {
+        Machine: {
+          Type: 'AWS::StepFunctions::StateMachine',
+          Properties: {
+            DefinitionString: {
+              'Fn::Join': [
+                '',
+                [
+                  '{ "Prop" : "new-value" }',
+                ],
+              ],
+            },
+          },
+        },
+      },
+    },
+  });
+
+  // WHEN
+  setup.pushStackMocks(setup.stackSummaryOf('Machine', 'AWS::StepFunctions::StateMachine', 'mock-machine-resource-id'));
+  const deployStackResult = await setup.tryHotswapDeployment({}, cdkStackArtifact);
+
+  // THEN
+  expect(deployStackResult).not.toBeUndefined();
+  expect(mockUpdateMachineDefinition).toHaveBeenCalledWith({
+    definition: '{ "Prop" : "new-value" }',
+    stateMachineArn: 'mock-machine-resource-id', // the sdk will convert the ID to the arn in a production environment
+  });
+});
+
+test('does not call the updateStateMachine() API when it receives a change to a parameter that is not the definitionString in a state machine', async () => {
+  // GIVEN
+  setup.setTemplate({
+    Resources: {
+      Machine: {
+        Type: 'AWS::StepFunctions::StateMachine',
+        Properties: {
+          DefinitionString: {
+            'Fn::Join': [
+              '',
+              [
+                '{ "Prop" : "old-value" }',
+              ],
+            ],
+          },
+          LoggingConfiguration: { // non-definitionString property
+            IncludeExecutionData: true,
+          },
+        },
+      },
+    },
+  });
+  const cdkStackArtifact = setup.cdkStackArtifactOf({
+    template: {
+      Resources: {
+        Machine: {
+          Type: 'AWS::StepFunctions::StateMachine',
+          Properties: {
+            DefinitionString: {
+              'Fn::Join': [
+                '',
+                [
+                  '{ "Prop" : "new-value" }',
+                ],
+              ],
+            },
+            LoggingConfiguration: {
+              IncludeExecutionData: false,
+            },
+          },
+        },
+      },
+    },
+  });
+
+  // WHEN
+  const deployStackResult = await setup.tryHotswapDeployment({}, cdkStackArtifact);
+
+  // THEN
+  expect(deployStackResult).toBeUndefined();
+  expect(mockUpdateMachineDefinition).not.toHaveBeenCalled();
+});
+
+test('does not call the updateStateMachine() API when a resource with type that is not AWS::StepFunctions::StateMachine but has the same properties is changed', async () => {
```

Comment:
This test name probably needs some polishing as well, doesn't feel like this sentence is correct 🙂.